### ISSUE TITLE
get_type_name hides subtypes for a union

### ISSUE THREAD

**Issue by GothAck** [agronholm/typeguard#97]
For example:
```py3
>>> typeguard.check_type('test var', [123], typing.Optional[typing.List[str]])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.7/dist-packages/typeguard/__init__.py", line 542, in check_type
    checker_func(argname, value, expected_type, memo)
  File "/usr/local/lib/python3.7/dist-packages/typeguard/__init__.py", line 394, in check_union
    format(argname, typelist, qualified_name(value)))
TypeError: type of test var must be one of (List, NoneType); got list instead
```

I see, at least in Python 3.7, __repr__ is defined for typing.* classes, would it be suitable to use the repr where available?

e.g.
```py3
>>> repr(typing.Optional[typing.List[str]])
'typing.Union[typing.List[str], NoneType]'
>>> repr(typing.List[str])
'typing.List[str]'
```

**Comment by agronholm** [comment]
The error message for a failing `Union` check is bad and issue #65 was set up to track the progress on the improvements (of which there are none to report yet).

**Comment by agronholm** [comment]
Fixed in master.

**Comment by Dobatymo** [comment]
> Fixed in master.

It's fixed in `3.0` branch not in `master` branch for anyone trying.
